### PR TITLE
Search widget issue fix

### LINKED ISSUE #625: [Bug] Search function with widget do not work properly in specific case.
### Are you ready to report?
    Yes 👍  

### Environment
  * version of the software you use : version name 2.3 (2018-03-12 master branch)
  * device / operating system : Samsung Galaxy S8, Huawei Honor A6
  
### The Bug:

  bug_report

In specific case when I use widget search, It does not go to search screen, goes to latest wiki page.

#### Steps to reproduce:

  1. Click widget's "search kiwix" button
  2. Then search screen will be appear. Press back press button to go to latest wiki page
  3. In latest wiki page, Press home button, go to home.
  4. At home, click widget's "search kiwix" button again.
  5. We'll see latest wiki page, not search screen.

If It work properly, at step 5 we can see search screen. But this case, It does not appear.

**I would like to work this issue.**

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes #625

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt [hunk 1]
```diff
@@ -114,10 +114,10 @@ class KiwixMainActivity : CoreMainActivity() {
     actionMode = mode
   }
 
-  override fun onNewIntent(intent: Intent) {
-    super.onNewIntent(intent)
+  override fun onNewIntent(newIntent: Intent) {
+    super.onNewIntent(newIntent)
     supportFragmentManager.fragments.filterIsInstance<FragmentActivityExtensions>().forEach {
-      it.onNewIntent(intent, this)
+      it.onNewIntent(newIntent, this)
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt:120]: We don't have to rename it.

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt [hunk 2]
```diff
@@ -155,10 +155,11 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
...
   }
 
-  override fun onNewIntent(intent: Intent) {
-    super.onNewIntent(intent)
+  override fun onNewIntent(newIntent: Intent) {
+    super.onNewIntent(newIntent)
+    intent.action = newIntent.action
     activeFragments().filterIsInstance<FragmentActivityExtensions>().forEach {
-      it.onNewIntent(intent, this)
+      it.onNewIntent(newIntent, this)
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt:162]: Here also.

Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt]: Done

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt (hunk 1 above)
Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt]: Done

**Review by gouri-panda — APPROVED** [review]
@MohitMaliFtechiz 🎉

**Pull request merged**